invenio_rdm_records/services/communities/service.py

```diff
@@ -231,3 +232,74 @@ def search(
             extra_filter=communities_filter,
             **kwargs
         )
+
+    @staticmethod
+    def _get_excluded_communities_filter(record, identity, id_):
+        """Return filter to exclude communities that should not be suggested."""
+        communities_to_exclude = []
+        communities_ids = record.parent.communities.ids
+
+        for community_id in communities_ids:
+            communities_to_exclude.append(dsl.Q("term", **{"id": community_id}))
+
+        open_requests = current_requests_service.search(
+            identity,
+            extra_filter=dsl.query.Bool(
+                "must",
+                must=[
+                    dsl.Q("term", **{"topic.record": id_}),
+                    dsl.Q("term", **{"type": CommunityInclusion.type_id}),
+                    dsl.Q("term", **{"is_open": True}),
+                ],
+            ),
+        )
+
+        for request in open_requests.hits:
+            communities_to_exclude.append(
+                dsl.Q("term", **{"id": request["receiver"]["community"]})
+            )
+
+        exclusion_filter = dsl.query.Bool("must_not", must_not=communities_to_exclude)
+
+        return exclusion_filter
+
+    def search_suggested_communities(
+        self,
+        identity,
+        id_,
+        params=None,
+        search_preference=None,
+        expand=False,
+        by_membership=False,
+        extra_filter=None,
+        **kwargs
+    ):
+        """Search for communities that can be added to a record."""
+        record = self.record_cls.pid.resolve(id_)
+
+        self.require_permission(identity, "add_community", record=record)
+
+        communities_filter = self._get_excluded_communities_filter(
+            record, identity, id_
+        )
+
+        if by_membership:
+            return current_communities.service.search_user_communities(
```

Comment:
@jrcastro2 here we are reusing the existing service of communities, with an extra filter to exclude communities - thanks to that we don't need to create another search method in this service